src/mlpack/methods/approx_kfn/approx_kfn_main.cpp

```diff
@@ -263,6 +263,15 @@ static void mlpackMain()
       arma::mat exactDistances;
       if (CLI::HasParam("exact_distances"))
       {
+        // Check the exact distances matrix has the right dimensions.
+        RequireParamValue<arma::mat>("exact_distances",
+          [&](arma::mat exactDists){ return (exactDists.n_rows == k); }, true,
+          "the number of rows in exact distances matrix must be equal to k.");
```

Comment:
I think it would be better to pass a reference to the lambda here to avoid copying the matrix:

```
[&](const arma::mat& exactDists) { ... }
```